src/settings.js

```diff
@@ -723,6 +723,7 @@ var SAFE_SPLIT_MEMORY = 0; // Similar to SAFE_HEAP, but for SPLIT_MEMORY.
 
 var RUNNING_JS_OPTS = 0; // whether js opts will be run, after the main compiler
 var BOOTSTRAPPING_STRUCT_INFO = 0; // whether we are in the generate struct_info bootstrap phase
+var STRUCT_INFO = ''; // struct_info that is either generated or cached
```

Comment:
These were being set in emscripten.py by writing to the settings dict. When using shared.Settings directly this warns that the setting is missing from settings.js. Before the `copy` call never checked that attrs were known to exist.